### PR TITLE
Add French translation

### PR DESCRIPTION
**Pull request by Huluti**
Generated by ChatGPT but seems a good start :)

### CODE REVIEW

**Comment by curry684** [comment]
I'll ask a Frenchie to double check, thanks  😉

File: translations/antispam+intl-icu.fr.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  banned_markup:
+    html: 'La valeur contient une balise HTML, ce qui n'est pas autorisé.'
+    bbcode: 'La valeur contient une balise BBCode, ce qui n'est pas autorisé.'
+
+  banned_phrases:
+    phrase_found: 'La valeur contient la phrase non autorisée "{phrase}".'
+
+  banned_scripts:
+    not_allowed: 'La valeur contient des caractères de scripts non autorisés ({scripts}).'
+    percentage_exceeded: 'La valeur contient {percentage}% de caractères de scripts non autorisés ({scripts}), alors que seuls {max}% sont autorisés.'
```
Comment (celinora) [inline translations/antispam+intl-icu.fr.yaml:24]: percentage_exceeded: 'La valeur est composée à {percentage}% de caractères provenant de scripts non autorisés ({scripts}). Le maximum autorisé est de : {max}%.'

_Note: the phrase with "alors que seuls {max}% sont autorisés." would be incorrect for 1% hence the change to make it more generic._

File: translations/antispam+intl-icu.fr.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    characters_exceeded: 'La valeur contient {count} caractères de scripts non autorisés ({scripts}), alors que seuls {max} sont autorisés.'
+
```
Comment (celinora) [inline translations/antispam+intl-icu.fr.yaml:26]: percentage_exceeded: 'La valeur contient {count} caractères provenant de scripts non autorisés ({scripts}). Le maximum autorisé est de : {max} caractère(s).'

_Note: the phrase with "alors que seuls {max} sont autorisés." would be incorrect for the value of max = 1 hence the change to make it more generic._

**Review by celinora — COMMENTED** [review]
Reviewed the french translation. See comments. Especially the part about "Caractères de script" was incorrect. I had to check the english and dutch translation to understand what it was about. The rest are minors changes but would look jut a little more native.

**Pull request merged**